app/models/pod_version.rb

```diff
@@ -73,16 +73,34 @@ def description
         "#{pod.name} #{name}"
       end
 
-      def push!(committer, specification_data)
-        update(:deleted => false)
-        response = PushJob.new(self, committer, specification_data).push!
+      def push!(committer, specification_data, change_type)
+        response = PushJob.new(self, committer, specification_data, change_type).push!
         if response.success?
+          update(:deleted => change_type == 'Delete')
           add_commit(:committer => committer, :sha => response.commit_sha, :specification_data => specification_data)
           pod.add_owner(committer) if pod.owners.empty?
         end
         response
       end
 
+      def deprecate!(committer, in_favor_of = nil)
+        raise "Can't deprecate a deleted spec: #{self}" if deleted?
+        spec = Specification.from_json(last_published_by.specification_data)
+        raise "Unable to find a podspec to deprecate: #{self}" unless spec
+        return if spec.deprecated?
```

Comment:
What I am wondering is: should it be possible to set the "deprecated in favor of" field even if the `spec.deprecated` field is true (in which case `spec.deprecated?` would be true and it would return).